posthog/api/test/test_decide.py

```diff
@@ -146,6 +146,82 @@ def test_feature_flags(self):
             self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(response.json()["featureFlags"], ["default-flag"])
 
+    def test_feature_flags_v2(self):
+        self.team.app_urls = ["https://example.com"]
+        self.team.save()
+        self.client.logout()
+        Person.objects.create(team=self.team, distinct_ids=["example_id"], properties={"email": "tim@posthog.com"})
+        FeatureFlag.objects.create(
+            team=self.team, rollout_percentage=50, name="Beta feature", key="beta-feature", created_by=self.user,
+        )
+        FeatureFlag.objects.create(
+            team=self.team,
+            filters={"groups": [{"properties": [], "rollout_percentage": None}]},
+            name="This is a feature flag with default params, no filters.",
+            key="default-flag",
+            created_by=self.user,
+        )  # Should be enabled for everyone
+        FeatureFlag.objects.create(
+            team=self.team,
+            filters={
+                "groups": [{"properties": [], "rollout_percentage": None}],
+                "multivariate": {
+                    "variants": [
+                        {"key": "first-variant", "name": "First Variant", "rollout_percentage": 50},
+                        {"key": "second-variant", "name": "Second Variant", "rollout_percentage": 25},
+                        {"key": "third-variant", "name": "Third Variant", "rollout_percentage": 25},
+                    ],
+                    "fallback_variant_key": "third-variant",
+                },
+            },
+            name="This is a feature flag with multiple variants.",
+            key="multivariate-flag",
+            created_by=self.user,
+        )
+
+        with self.assertNumQueries(2):
+            response = self._post_decide(api_version=1)  # v1 functionality should not break
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            self.assertIn("beta-feature", response.json()["featureFlags"])
+            self.assertIn("default-flag", response.json()["featureFlags"])
+            self.assertIn("multivariate-flag", response.json()["featureFlags"])
```

Comment:
Correct me if I'm wrong, but if we're using api v < 2, wouldn't we want multivariate-flag to not be part of the results?

Reply:
My thinking was to leave it in, as it's technically still "active" if the user qualifies for the filters and rollout group. That's similar to how I reason about it internally on models/feature_flag.py:196-200 for the new API.

```py
if feature_flag.distinct_id_matches(distinct_id):
    flags_enabled[feature_flag.key] = True
    if len(feature_flag.variants) > 0:
        variant = feature_flag.get_variant_for_distinct_id(distinct_id) or False
        flags_enabled[feature_flag.key] = variant
```

There may be a case where a user wants to check for the presence or absence of a multivariate flag.

I'm thinking of this as, partial support is better than none. _More_ data is available through v2 but it's not a completely separate type to the point where we shouldn't return it for v1.

Reply:
I think I have a slightly different read on the tradeoff here –

Thinking from the developer's standpoint, they have probably coded a few flows based on the variant strings.

The check they would use is probably presence of the flag in the dict that comes back – and downstream from this check, they would probably look for variant strings and be confused why the flag is returning but there are no variants.

So my feeling is the fact that there is partial support may be more confusing for the developer than having nothing returned at all – does that make sense?

I think it would be a lot simpler (from a consumer pov) to just not have it returned at all if they're consuming it with an API that doesn't support the use cases they defined themselves in PostHog.

Reply:
Sounds good. Will change.